core/state_transition.go

```diff
@@ -374,27 +373,29 @@ func (st *StateTransition) TransitionDb() (ret []byte, usedGas uint64, failed bo
 		return nil, 0, false, err
 	}
 
-	// Pay tx fee to tx fee recipient and the proposer (coinbase)
-	txFeeRecipient := msg.GasFeeRecipient()
-	if txFeeRecipient == nil {
-		sender := msg.From()
-		txFeeRecipient = &sender
-	}
-
 	gasUsed := new(big.Int).SetUint64(st.gasUsed())
 	totalTxFee := new(big.Int).Mul(gasUsed, st.gasPrice)
-	// gasUsed * gasPriceMinimun * proposerFractionNumerator / proposerFractionDenominator
-	coinbaseTxFee := new(big.Int).Div(new(big.Int).Mul(gasUsed, new(big.Int).Mul(st.gasPriceMinimum, st.proposerFraction.Numerator)), st.proposerFraction.Denominator)
-	recipientTxFee := new(big.Int).Sub(totalTxFee, coinbaseTxFee)
+	// Divide the transaction into a base (the minimum transaction fee) and tip (any extra).
+	baseTxFee := new(big.Int).Mul(gasUsed, st.gasPriceMinimum)
+	tipTxFee := new(big.Int).Sub(totalTxFee, baseTxFee)
 
-	if err = st.creditGas(st.evm.Coinbase, coinbaseTxFee, msg.GasCurrency()); err != nil {
+	if err = st.creditGas(st.evm.Coinbase, tipTxFee, msg.GasCurrency()); err != nil {
 		return nil, 0, false, err
 	}
-	if err = st.creditGas(*txFeeRecipient, recipientTxFee, msg.GasCurrency()); err != nil {
-		return nil, 0, false, err
+
+	governanceAddress, err := vm.GetRegisteredAddressWithEvm(params.GovernanceRegistryId, evm)
+	if err != nil {
+		log.Trace("Cannot credit gas fee to infrastructure fund: burning the fee", "error", err, "fee", baseTxFee)
```

Comment:
My concern here is that burning the fee would break the invariant that `totalSupply` is the sum of balances over all accounts. Which I suppose nothing relies on strictly, but it makes me uneasy.

Is there somewhere else we could send it that would make sense?

Reply:
I see two options here:
1.  Send it to either the sender or the block proposer, both of which expose the vulnerability we are trying to eliminate, but only until the Governance contract is deployed.
2. Have a fixed address (like out `ce10` address for the registry) or one that is configured in the genesis block act as the account for the infrastructure fund so we know it is always available.

Reply:
+1 to option 1, I think it's fine for this vulnerability to exist in that small window during network initialization. There even exists a world in which we encode the deployment of all the core contract into the genesis block.

I don't have a strong preference for sending it to the proposer vs the sender. For the former, we expose the attack where the block proposers will want to fill the block with fake txs to drive up fee revenue. For the latter, we expose the DOS risk of potentially free txs for users, which only occurs if tx senders specify the exact gas price minimum as their gas price. In the case where there's competition, users will need to specify higher gas prices, and I think this actually winds up reducing to be more or less equivalent to the market for txs in ethereum (though not exactly). So actually, would probably suggest sending it back to the sender.

Reply:
Sending the fees to the sender has the same gas price minimum (GPM) attack as if it was sent to proposer because the proposer can construct a number of empty transactions that are send from accounts they control, so it has both the GPM attack risk and the DOS attack risk.
One addition to this PR that could be made is to only enforce the GPM if the governance address is available in addition to the GPM parameter. That will encode the current Ethereum market system as the fallback and eliminate this window for a GPM attack to occur (as there will be no enforce GPM to push up)

Reply:
We discussed this offline: My assertion that the GPM attack is still feasible in the "refund to sender" scenario is not true because the proposers will have no incentive to push the GPM up (and a weak incentive to push it down as a high GPM in that scenario means the caller needs a certain amount of funds in their account)
We will go with the "refund to sender" approach